tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -36,161 +32,29 @@ public void run() {
     public void process(final Socket connection) {
```

Comment:
```suggestion
HttpResponse response = HttpResponse.init();
            HttpRequest request = new HttpRequest(inputStream);
            try {
                requestHandler.handle(request, response);
            } catch (IOException e) {
                log.error(e.getMessage(), e);
                byte[] body = Files.readAllBytes(get400ErrorPage());
                response.status(400).
                        .header("Content-Type", ContentType.TEXT_HTML.getMimeType())
                        .header("Content-Length", String.valueOf(body.length))
                        .body(body);
            }
```

tomcat에서는 Processor가 Request와 Response의 객체를 생성하고 Handler로 넘깁니다.
즉 Processor가 Response와 Request의 생명주기를 관리하고 Handler는 Response 데이터를 쓰는 책임을 가집니다.

지금 구조는 RequestHandler가 Response를 생성하고 데이터까지 쓰는 책임을 가지는데요, 물론 간단한 구조라서 이렇게 해도 충분하지만 tocmat의 구조의 따라가보는 것도 좋은 경험이라고 생각합니닷

알고 넘어가신거면 굳이 반영안하셔도됩니당!!~

Reply:
맞아요👍 LMS에 주어진 코드도 `requestHandler.handle(request, response);` 이런 형태더라고요.
말씀해주신 것처럼 지금과 같이 간단한 구조에서는 불필요해 보여서 Handler에서 Response를 생성하도록 구현했습니다!

Reply:
톰캣이 그런 구조를 갖는 이유가 궁금해서 따로 찾아보았는데요, 응답 전까지 헤더/상태 코드 변경이 자유롭고, 스트림 처리로 대용량 데이터 전송 시 메모리를 아낄 수 있는 것이 장점이라고 합니다!

사실 크게 와닿지 않아서 가콩이 더 알고계신게 있는지도 궁금하네요~🧐

Reply:
넹 저도 공부해봤었을 때 짱구가 말한 유연한 응답, 스트림으로 메모리에 올리지 않아도 되는 점,
그리고 Processor가 Request, Resposne의 생명주기를 관리하고, Handler는 응답을 작성에 집중한ㄷ나는 세가지 이유가 잇는 것 같아욤~

특히 유연한 응답과, 책임 분리 관점에서 좀 더 살펴보자면 
실제 WebApplication에서는 하나의 Response를 만들기 위해 여러 Handler가 함께 동작합니다. (filter 같은 애) 
만약 Handler들이 직접 Response를 생성 및 반환하면 하나의 Reqeust에 대해 Reponse가 여러개 생기거나, 생성의 책임도 모호해질 것 같아요! 

따라서 요청을 받고 응답을 보내는 상위 계층의 Processor가 Response를 한번만 생성하고 Handler는 그 위에 작성해서 
결국 짱구가 말한 유연한 응답을 할 수 있는 것 아닐까요?

Reply:
아하, 자세한 설명 덕분에 궁금했던 부분이 명확하게 해소되었어요😊  감사합니다 가콩~